include/aws/common/private/atomic.inl

```diff
@@ -0,0 +1,98 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#if defined(__GNUC__) || defined(__clang__)
+
+    static inline int aws_atomic_get(int *dst) {
+        int value;
+        do {
+            value = *dst;
+        } while (!aws_atomic_compare_exchange(dst, value, value));
+        return value;
+    }
+
+    static inline int aws_atomic_set(int *dst, int value) {
+        return __sync_lock_test_and_set(dst, value);
```

Comment:
this adds a minimum compiler version for intel. Do we want to support intel compilers explicitly?

Reply:
I don’t really know how to answer this. I’m unfamiliar with intel compilers and whatnot.